packages/i18n/src/sprintf.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+import _sprintf from '@tannin/sprintf';
+import type { SprintfArgs } from '@tannin/sprintf/types';
+
+/**
+ * Internal dependencies
+ */
+import type { TranslatableText } from './types';
+
+type DistributeSprintfArgs< T extends string > = T extends any
+	? SprintfArgs< T >
+	: never;
+
+export function sprintf< T extends string >(
+	format: T | TranslatableText< T >,
+	...args: DistributeSprintfArgs< T >
+): string;
+export function sprintf< T extends string >(
+	format: T | TranslatableText< T >,
+	args: DistributeSprintfArgs< T >
+): string;
+
+/**
+ * Returns a formatted string.
+ *
+ * @template {string} T
+ * @param {T | TranslatableText<T>} format The format of the string to generate.
+ * @param {SprintfArgs<T>}          args   Arguments to apply to the format.
```

Comment:
This can also be updated to `DistributeSprintfArgs< T >`
```suggestion
 * @param {DistributeSprintfArgs<T>}          args   Arguments to apply to the format.
```